crypto/keys/signer_factory_tester.go

```diff
@@ -65,22 +63,20 @@ func (tester *SignerFactoryTester) RunAllTests(t *testing.T) {
 func (tester *SignerFactoryTester) TestNewSigner(t *testing.T) {
 	for _, test := range append(tester.NewSignerTests, []NewSignerTest{
 		{
-			Name: "Unsupported PrivateKey type",
-			Tree: &trillian.Tree{
-				PrivateKey: mustMarshalAny(&empty.Empty{}),
-			},
-			WantErr: true,
-		},
-		{
-			Name:    "No PrivateKey",
```

Comment:
We don't typically check for nil pointers being passed into funcs. I'm happy to add this if it's a convention we want to start following. I've heard people argue in favour of crashing on nil pointers, but returning an error seems better to me.